pulsar-common/src/main/java/org/apache/pulsar/common/util/SimpleTextOutputStream.java

```diff
@@ -44,7 +46,8 @@ public SimpleTextOutputStream write(byte[] a, int offset, int len) {
     }
 
     public SimpleTextOutputStream write(char c) {
-        write(String.valueOf(c));
+        singleCharBuffer.put(0, c);
+        buffer.writeCharSequence(singleCharBuffer, CharsetUtil.UTF_8);
```

Comment:
This change reduces a lot of String allocations. This was visible in allocation profiling.

Reply:
👍 I think the original `writeByte()` was appropriate here for the single char. This is only used for `{`, `'` and `"` characters, it will never have any non ascii chars

Reply:
@merlimat good point. I'll add the writeByte optimization for the ascii character range where it's safe.

Reply:
@merlimat optimization added in bf1e994